### PR TITLE
fix: Skip echarts rerendering if props are same

### LINKED ISSUE #25783: [Bug]: The widget gets reset and displays the hidden series chart as well on clicking on chart datapoints
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

This issue seen after updating charts from fusion to ECharts

### Steps To Reproduce

1. Drag & Drop a Chart widget
2. Add multi series chart data
3. Click some legends to hide the series data graph
4. Click on the datapoints visible on chart
5. Observe that the hidden chart graphs are visible now

### Public Sample App

_No response_

### Environment

Production

### Issue video log

https://www.loom.com/share/d0545e25480e4fcc8c1fbdbc4bb5c8d5?sid=64fcbf2b-05a1-4044-8906-33c05933a854

### Version

Cloud/Release

### PR DESCRIPTION
**Pull request by rajatagrawal**
Fixes #25783

### CODE REVIEW

File: app/client/src/widgets/ChartWidget/component/index.tsx [hunk 1]
```diff
@@ -177,19 +177,12 @@ class ChartComponent extends React.Component<
...
-          this.props.dimensions.componentHeight ||
-        this.echartsInstance.getWidth() != this.props.dimensions.componentWidth
-      ) {
-        this.echartsInstance.resize({
-          width: this.props.dimensions.componentWidth,
-          height: this.props.dimensions.componentHeight,
-        });
-      }
-    }
+  echartsResizeNeeded = () => {
```
Comment (sbalaji1192) [inline app/client/src/widgets/ChartWidget/component/index.tsx:180]: lets change name of this function to `shouldResize`

File: app/client/src/widgets/ChartWidget/component/index.tsx [hunk 2]
```diff
@@ -200,33 +193,31 @@ class ChartComponent extends React.Component<
...
       this.echartConfiguration = newConfiguration;
-    }
-
-    try {
-      this.echartsInstance.off("click");
-      this.echartsInstance.on("click", this.dataClickCallback);
 
-      if (needsNewConfig) {
+      try {
+        this.echartsInstance.off("click");
```
Comment (sbalaji1192) [inline app/client/src/widgets/ChartWidget/component/index.tsx:203]: nit: let's move out the lines that will not throw errors from the try block

File: app/client/src/widgets/ChartWidget/component/index.tsx (hunk 1 above)
Comment (rajatagrawal) [reply app/client/src/widgets/ChartWidget/component/index.tsx:180]: Done

File: app/client/src/widgets/ChartWidget/component/index.tsx (hunk 2 above)
Comment (rajatagrawal) [reply app/client/src/widgets/ChartWidget/component/index.tsx]: Done

**Review by sbalaji1192 — APPROVED** [review]

**Comment by chandannkumar** [comment]
Tested this PR and LGTM

**Pull request merged**